scipy/sparse/linalg/_dsolve/SuperLU/scipychanges.patch

```diff
@@ -413,15 +358,82 @@ index 351e0ad62..4015b2380 100644
  	zgemv_( "N", &nrow, &nsupc, &alpha, &lusup[luptr+nsupc], &nsupr, 
-diff --git a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_scomplex.h b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_scomplex.h
-index 40c22924c..83be8c971 100644
---- a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_scomplex.h
-+++ b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_scomplex.h
-@@ -30,7 +30,6 @@ at the top-level directory.
+diff --git a/scipy/sparse/linalg/dsolve/SuperLU/SRC/scipy_slu_config.h b/scipy/sparse/linalg/dsolve/SuperLU/SRC/scipy_slu_config.h
+new file mode 100644
+index 000000000..5afc93b5d
+--- /dev/null
++++ b/scipy/sparse/linalg/dsolve/SuperLU/SRC/scipy_slu_config.h
+@@ -0,0 +1,36 @@
++#ifndef SCIPY_SLU_CONFIG_H
++#define SCIPY_SLU_CONFIG_H
++
++#include <stdlib.h>
++
++/*
++ * Support routines
++ */
++void superlu_python_module_abort(char *msg);
++void *superlu_python_module_malloc(size_t size);
++void superlu_python_module_free(void *ptr);
++
++#define USER_ABORT  superlu_python_module_abort
++#define USER_MALLOC superlu_python_module_malloc
++#define USER_FREE   superlu_python_module_free
++
++#define SCIPY_FIX 1
++
++/*
++ * Fortran configuration
++ */
++#if defined(NO_APPEND_FORTRAN)
++#if defined(UPPERCASE_FORTRAN)
++#define UpCase 1
++#else
++#define NoChange 1
++#endif
++#else
++#if defined(UPPERCASE_FORTRAN)
++#error Uppercase and trailing slash in Fortran names not supported
++#else
++#define Add_ 1
++#endif
++#endif
++
++#endif
+diff --git a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_Cnames.h b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_Cnames.h
+index b95c01830..97ba246cc 100644
+--- a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_Cnames.h
++++ b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_Cnames.h
+@@ -29,6 +29,7 @@ at the top-level directory.
+ #ifndef __SUPERLU_CNAMES /* allow multiple inclusions */
+ #define __SUPERLU_CNAMES
+ 
++#include "scipy_slu_config.h"
  
- typedef struct { float r, i; } singlecomplex;
+ #define ADD_       0
+ #define ADD__      1
+diff --git a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_util.h b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_util.h
+index 1afbdd0e7..61552b753 100644
+--- a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_util.h
++++ b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/slu_util.h
+@@ -32,6 +32,7 @@ at the top-level directory.
+ #include <assert.h>
+ #include "superlu_enum_consts.h"
+ 
++#include "scipy_slu_config.h"
  
--#define complex singlecomplex  // backward compatibility
+ /***********************************************************************
+  * Macros
+diff --git a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/superlu_config.h b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/superlu_config.h
+index eeb2a1e43..c3e7bac1c 100644
+--- a/scipy/sparse/linalg/_dsolve/SuperLU/SRC/superlu_config.h
++++ b/scipy/sparse/linalg/_dsolve/SuperLU/SRC/superlu_config.h
+@@ -3,7 +3,7 @@
+ #define SUPERLU_CONFIG_H
  
- /* Macro definitions */
+ /* Enable metis */
+-#define HAVE_METIS TRUE
++/*#define HAVE_METIS FALSE */
```

Comment:
Commenting this out is necessary. Otherwise, at runtime we get errors about missing METIS symbols.

Reply:
This change seems fine. Enabling this unconditionally is probably not a good thing to do upstream (that's what build system checks are for), but let's not worry about it too much for now since the patch is harmless.

Reply:
I thought the same. Unfortunately, setting it to FALSE is also not enough. Commented upstream here: https://github.com/xiaoyeli/superlu/commit/6f7ea89bce984b61420cf9c6c1a89275e68068c7#r145014670